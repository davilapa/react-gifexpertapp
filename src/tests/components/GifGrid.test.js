import { GifGrid } from "../../components/GifGrid";
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')
describe('GifGrid tests', () => {
    const category = 'Goku';

    test('should render GifGrid correctly', () => { 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid  category={category}/>);

        expect(wrapper).toMatchSnapshot();
     });

    test('should show items when images are loading useFetchGifs', () => { 
        useFetchGifs.mockReturnValue({
            data: [
                {
                    id: '1',
                    title: 'Goku',
                    url: 'goku.com'
                }
            ],
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(1);
     });


});