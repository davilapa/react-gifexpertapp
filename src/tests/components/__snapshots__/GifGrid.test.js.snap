// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render GifGrid correctly 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
